pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
 
+	var experimentalClient *client.ExperimentalClient
+	cfg, experimentalClientErr := clientConfig.ClientConfig()
```

Comment:
You can't do this here - it fails too early (when factory is created, not when methods on factory are called).  Initialization like this should always be under a Factory method (so that errors are returned when the user asks for clients, NOT when the user asks for a factory (clients are not required to be valid for all command types).